packages/react/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,114 @@
+import { Slot } from '@radix-ui/react-slot';
+import cl from 'clsx/lite';
+import {
+  Fragment,
+  type HTMLAttributes,
+  type ReactNode,
+  forwardRef,
+  useMemo,
+} from 'react';
+
+export type AvatarProps = {
+  /**
+   * The name of the person the avatar represents.
+   * Will be used to generate initials if no children are provided.
+   */
+  name?: string;
```

Comment:
Late to to party here 😅

Why do we want a separate prop for generating initials? Is it not simpler for people to just define it themselves in children? I know its optional, but seems just like an additional thing we have to maintain that we really maybe shouldn't give how our other components are as simple as it gets? 🤔

Reply:
We need the name for labeling :)

Reply:
I feel it's safer for us to do this, rather than rely on consumers to set a prop that is highly necessary

Reply:
Just felt it was a bit counter intuitive that I couldn't do ` <Avatar>ON</Avatar>` for deciding my "own" initials. As its now passed text to children is ignored.

Reply:
You can still pass text down, but it needs to be wrapped in an element. This has nothing to do with the name prop, but rather that we are arrind `aria-hidden` with `Slot`, which needs an element

Reply:
If we are going to have an opinonated way of genareting initals could we also not support if they want to define their own (with `aria-label` ofc)?

Reply:
I can add a story about this. We don't want any text inside of the avatar to be read out, hence the `aria-hidden`.
You can also override our `aria-label` by adding your own. 
This is valid:
```
<Avatar name="Ole" aria-label="someone">
 <span>SK</span>
</Avatar>

Reply:
Can we make it work without the inne `<span>`? 🤔 
If adding:
```
const Component = children && typeof children !== 'string' ? Slot : Fragment;
```
Just doing:
```
<Avatar name="Ole">
  OJ
</Avatar>
```
will work ☺️

Reply:
Doing `aria-label` as required instead of `name` is would be kind of cool - supporting this ☺️

Reply:
> Can we make it work without the inne `<span>`? 🤔 If adding:
> 
> ```
> const Component = children && typeof children !== 'string' ? Slot : Fragment;
> ```
> 
> Just doing:
> 
> ```
> <Avatar name="Ole">
>   OJ
> </Avatar>
> ```
> 
> will work ☺️

We are already doing something similar with `Combobox`, so I don't see a reason as to why we can not do this here as well :) 

---

> Doing `aria-label` as required instead of `name` is would be kind of cool - supporting this ☺️

Sure, but we should drop generating initials for them then, and we need to document that it should be a maximum of 2 letters

Reply:
Why would generating initials change when moving to aria-label - we can use values from this as well? ☺️

Reply:
Suggest we drop the `name` prop and maybe flag `aria-label` as required? 

This will make the component more pure and closer to how a simple HTML+CSS markup would be. We always add the `name` functionality later :) 

It feels a bit weird to flag native props are required, but is there any use-case where it makes sense to not use `aria-label`, alternatively `aria-labeledby`, but that might get abused or prone to error 🤔

Reply:
Okay, so drop the initials generation as well? Should we export the function we have? Could make it a hook